packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -68,12 +69,22 @@ class RunCommand extends RunCommandBase {
         defaultsTo: false,
         negatable: false,
         help: 'Start in a paused mode and wait for a debugger to connect.');
+    argParser.addFlag('pub',
+        defaultsTo: true,
+        help: 'Whether to run "pub get" before running the app.');
     argParser.addOption('debug-port',
         defaultsTo: observatoryDefaultPort.toString(),
         help: 'Listen to the given port for a debug connection.');
   }
 
   @override
+  Future<int> run() async {
+    if (argResults['pub'])
+      await pubGet();
```

Comment:
That's too late.  We will have already decided this working directory is not a valid project because it doesn't have a sky_engine package.